src/routes/donations.js

```diff
@@ -197,26 +210,77 @@ function _validateIncomingDonation(incoming) {
  * created and then updated in the same upload,
  * they are marked as a *new* donor.
  */
-function _dedupeDonors(results) {
-  return uniqBy(results, donor => donor.id && donor.__isNew).reduce(
-    (donors, donor) => {
-      const idx = donors.findIndex(d => d.id === donor.id)
-      if (idx < 0) {
-        // have not encountered this donor yet
-        donors.push(donor)
-      } else if (donors[idx].__isNew && !donor.__isNew) {
-        // existing is new, incoming is updated:
-        // take updated details and mark as new
-        donors[idx] = donor
-        donors[idx].__isNew = true
+
+function _groupDonors(results) {
+  const groupById = _.groupBy(results, 'id')
+  const groupedArr = _.map(groupById, (details, id) => {
+    const donationCount = details.length
+    const sum = details.reduce((sum, donation) => {
+      return sum + donation.donationAmount
+    }, 0)
+    const name = _.last(details).name
+    const idNo = _.last(details).idNo
+    const __isNew = _.first(details).__isNew
+
+    return {
+      id,
+      idNo,
+      name,
+      totalAmount: sum,
+      donationCount,
+      __isNew
+    }
+  })
+
+  return groupedArr
+}
+
+function summary(results) {
+  const donations = _.map(results, el => el.donationAmount)
+  const totalAmt = donations.reduce((sum, donation) => {
+    return sum + donation
+  }, 0)
+  const totalCount = results.length
+  const dateFormatter = _.map(results, el => Date.parse(el.donationDate))
+  const maxDate = dateStringOf(new Date(Math.max.apply(null, dateFormatter)))
+  const minDate = dateStringOf(new Date(Math.min.apply(null, dateFormatter)))
```

Comment:
Normally, we will leave the responsibility of functions like `dateStringOf` to the front end, so that they can handle for different languages and formats. For now I think it's fine though.

Reply:
ok! I will work change it after the merge?